File: slack-reader/src/main/java/dev/number6/slackreader/dagger/SlackPortModule.java
package dev.number6.slackreader.dagger;

import dev.number6.slack.port.HttpPort;
import dev.number6.slackreader.adaptor.SlackReaderAdaptor;
import dev.number6.slackreader.port.SlackPort;
import dagger.Module;
import dagger.Provides;

@Module
public class SlackPortModule {

    @Provides
    public SlackPort slackClient(HttpPort httpPort) {
        return new SlackReaderAdaptor(httpPort);
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/dagger/SlackPortModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.port.HttpPort
import dev.number6.slackreader.adaptor.SlackReaderAdaptor

import dev.number6.slackreader.port.SlackPort

@Module
class SlackPortModule {
    @Provides
    fun slackClient(httpPort: HttpPort): SlackPort {
        return SlackReaderAdaptor(httpPort)
    }
}